GDJS/Runtime/variable.ts

```diff
@@ -194,7 +194,14 @@ namespace gdjs {
           return obj;
         case 'array':
           const arr: any[] = [];
-          for (const item of this._childrenArray) arr.push(item.toJSObject());
+          for (const item of this._childrenArray) {
+            // item can be undefined if there are empty elements in the array.
+            // A variable can have empty items in its children array if one inserts
+            // a variable at an index greater than highest index. All the array elements
+            // in the gap will be empty elements.
+            if (item === undefined) arr.push(undefined);
```

Comment:
can `null` be also a value of the array?

Reply:
It shouldn't be. It's targeted against this use-case:
```
// variable is an array variable with 2 children at indices 0 and 1
variable.getChildAt(5)
// variable is now [XXX, YYY, empty, empty, empty, ZZZ]
```